lib/grp.gi

```diff
@@ -1013,6 +1013,32 @@ RedispatchOnCondition( FittingSubgroup, true, [IsGroup], [IsFinite], 0);
 ##
 #M  FrattiniSubgroup( <G> ) . . . . . . . . . .  Frattini subgroup of a group
 ##
+InstallMethod( FrattiniSubgroup, "generic method for nilpotent groups",
+            [ IsGroup ],
+            RankFilter( IsGroup and IsNilpotentGroup )
+            - RankFilter( IsGroup ),
+function(G)
+local i, p, q, gen, Gf;
+    if IsTrivial(G) then
+      return G;
+    elif IsAbelian(G) then
+        gen := [ ];
+        for i in [1..Length(AbelianInvariants(G))] do
+            q := AbelianInvariants(G)[i];
+            if q<>0 and not IsPrime(q) then
+                p := SmallestRootInt(q);
+                Add(gen, IndependentGeneratorsOfAbelianGroup(G)[i]^p);
+            fi;
```

Comment:
@fingolfin Ok, now I call them in the beginning and then use the elements of this stored list. 
The reason I did not want to do this was that if an abelian group has only invariants of 0 and prime then we would not even need to call IndependentGenerators, which could be more expensive than AbelianInvariants.